### PR TITLE
add profile name to post and comment APIs

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

#188 

#### What's this PR do?

this adds the `.name` from the user's profile to the results returned by both the comment and post APIs, and makes the changes in the UI to display that information.

#### How should this be manually tested?

If you look at a post or a channel you should see names instead of usernames, same with comments on a post.

### CODE REVIEW

**Comment by rhysyngsun** [comment]
I think this PR needs to be rebased/refactored off @noisecapella's PR #198 once that merges so that we can accoutn for cases where `User` doesn't exist.

**Comment by aliceriot** [comment]
@rhysyngsun ok I have rebased over George's changes and incorporated the newer pattern for including profile information on the responses. I also added `.select_related('profile')` when we grab the users, to avoid making another query on each call to the serializer (for the post and comment list views) to grab the profile information.

File: channels/serializers.py [hunk 1]
```diff
@@ -144,6 +145,13 @@ def get_url(self, instance):
         """Returns a url or null depending on if it's a self post"""
         return instance.url if not instance.is_self else None
 
+    def get_author_name(self, instance):
+        """get the authors name"""
+        user = self._get_user(instance)
+        if user and user.profile.name:
+            return user.profile.name
+        return "Someone"
```
Comment (rhysyngsun) [inline channels/serializers.py]: Could you change this to be "[deleted]" to preserve the original default value?

File: channels/serializers.py [hunk 2]
```diff
@@ -290,6 +299,15 @@ def get_profile_image(self, instance):
...
         return default_profile_image
 
+    def get_author_name(self, instance):
+        """get the author name"""
+        # this is a similar setup to the get_profile_image thingy above
+        user = self._get_user(instance)
+
+        if user and user.profile.name:
+            return user.profile.name
+        return "Someone"
```
Comment (rhysyngsun) [inline channels/serializers.py]: Same comment as above

File: static/js/components/PostDisplay.js [hunk 3]
```diff
@@ -115,7 +115,7 @@ class PostDisplay extends React.Component {
             {postTitle(post)}
           </div>
           <div className="authored-by">
-            <a>{post.author_id || "someone"}</a> {formattedDate}{" "}
+            <a>{post.author_name || "someone"}</a> {formattedDate}{" "}
```
Comment (rhysyngsun) [inline static/js/components/PostDisplay.js]: This will never be null now, so we don't need the default.

**Comment by rhysyngsun** [comment]
Functionality looks good.

File: channels/serializers.py (hunk 1 above)
Comment (aliceriot) [reply channels/serializers.py]: sure

**Comment by aliceriot** [comment]
@rhysyngsun I believe I address all your comments

**Pull request merged**